### PR TITLE
Add dir support to kando with kopia location

### PR DESCRIPTION
**Pull request by PrasadG193**
Signed-off-by: Prasad Ghangal <prasad.ghangal@gmail.com>

## Change Overview

This approach has the following limitations:
1. The backup performed with stdin as a source, can be restored ONLY as stdout target
2. The backup performed with file/dir path as a source, can be restored ONLY as file/dir path as target

Please see the test plan for the details.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [x] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Test Plan

- [x] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

Test cases:
1. Backup file with stdin as source, restore file with stdout as target - PASS
2. Backup file with stdin as source, restore file with filepath as target - FAIL
3. Backup file with filepath as source, restore file with filepath as target - PASS
4. Backup file with filepath as source, restore file with stdout as target - FAIL
5. Backup dir with filepath as source, restore file with stdout as target and `--path` arg with same value as source filepath  - PASS
6. Backup dir with dirpath as source, restore dir with dirpath as target - PASS

### CODE REVIEW

**Comment by pavannd1** [comment]
I think test case 2 should work if the target path is same as the backup file path. Please try it out.

File: pkg/kopia/stream.go [hunk 1]
```diff
@@ -114,6 +117,78 @@ func Write(ctx context.Context, path string, source io.Reader) (*SnapshotInfo, e
 	return snapshotInfo, nil
 }
 
+func WriteFile(ctx context.Context, path string, sourcePath string) (*SnapshotInfo, error) {
```
Comment (pavannd1) [inline pkg/kopia/stream.go:120]: Please add a comment

File: pkg/kopia/stream.go [hunk 2]
```diff
@@ -146,6 +221,42 @@ func Read(ctx context.Context, backupID, path string, target io.Writer) error {
 	return errors.Wrap(err, "Failed to copy snapshot data to the target")
 }
 
+func ReadFile(ctx context.Context, backupID, target string) error {
```
Comment (pavannd1) [inline pkg/kopia/stream.go:224]: Same here. Requires a comment

**Comment by PrasadG193** [comment]
> I think test case 2 and 4 should work if the target path is same as the backup file path. Please try it out.

@pavannd1 it works on master where we create Reader/Writer from file and pass it to stream handling. But since we have added separate handling for files and streaming data, they are no longer compatible.

File: pkg/kopia/stream.go [hunk 3]
```diff
@@ -146,6 +221,42 @@ func Read(ctx context.Context, backupID, path string, target io.Writer) error {
...
+	if err != nil {
+		return errors.Wrap(err, "Unable to resolve path")
+	}
+	// TODO: Do we want to keep this flags configurable?
+	output := &restore.FilesystemOutput{
+		TargetPath:             p,
+		OverwriteDirectories:   true,
+		OverwriteFiles:         true,
+		OverwriteSymlinks:      true,
+		IgnorePermissionErrors: true,
```
Comment (PrasadG193) [inline pkg/kopia/stream.go:251]: @pavannd1 do we want to have these flags configurable?

Comment (pavannd1) [reply pkg/kopia/stream.go:251]: Should be okay for now. We can improve later if the requirement comes up.

**Review by pavannd1 — APPROVED** [review]
LGTM

**Pull request merged**